cruft/_commands/diff.py

```diff
@@ -0,0 +1,62 @@
+import json
+import shutil
+import sys
+from pathlib import Path
+from tempfile import TemporaryDirectory
+
+import typer
+
+from . import _diff, _generate
+from .utils import get_cookiecutter_repo, get_cruft_file
+
+
+def diff(project_dir: Path = Path("."), exit_code: bool = False) -> bool:
+    """Show the diff between the project and the linked Cookiecutter template"""
+    cruft_file = get_cruft_file(project_dir)
+    cruft_state = json.loads(cruft_file.read_text())
+    checkout = cruft_state.get("commit")
+
+    has_diff = False
+    with TemporaryDirectory() as tmpdir_:
+        tmpdir = Path(tmpdir_)
+        repo_dir = tmpdir / "repo"
+        remote_template_dir = tmpdir / "remote"
+        local_template_dir = tmpdir / "local"
+
+        # Create all the directories
+        remote_template_dir.mkdir(parents=True, exist_ok=True)
+        local_template_dir.mkdir(parents=True, exist_ok=True)
+
+        # Let's clone the template
+        repo = get_cookiecutter_repo(cruft_state["template"], repo_dir)
+
+        # We generate the template for the revision expected by the project
+        _generate.cookiecutter_template(
+            output_dir=remote_template_dir, repo=repo, project_dir=project_dir, checkout=checkout
+        )
+
+        # Then we create a new tree with each file in the template that also exist
+        # locally.
+        for path in sorted(remote_template_dir.glob("**/*")):
+            relative_path = path.relative_to(remote_template_dir)
+            local_path = project_dir / relative_path
+            destination = local_template_dir / relative_path
+
+            if path.is_file():
+                shutil.copy(str(local_path), str(destination))
+            else:
+                destination.mkdir(parents=True, exist_ok=True)
+                destination.chmod(local_path.stat().st_mode)
+
+        # Finally we can compute and print the diff.
+        diff = _diff.get_diff(remote_template_dir, local_template_dir)
+
+        if diff.strip():
+            has_diff = True
+
+            if exit_code or not sys.stdout.isatty():
```

Comment:
git diff should already take care of this by itself? Do we still need to check for isatty?

Reply:
> Do we still need to check for isatty?

Yes, we need it because running `git diff` directly prefixes paths with `/tmp/xyz`. If the terminal isn't a TTY, we don't want this. (e.g. doing `cruft diff > cruft.patch` to apply the patch later on or make changes to it)

Reply:
I'll add a comment to make it clear.

Reply:
Done.